File: experiments_past_logs/RQ1_baxbench/_INDICT/baxbench_gpt4o/indict_llama_round3/gpt-4o/PDFToText/PHP-Lumen/temp0.0-openapi-none/sample0/code/app/Http/Controllers/PDFController.php
<?php

namespace App\Http\Controllers;

use Illuminate\Http\Request;
use Symfony\Component\HttpFoundation\Response;

class PDFController extends Controller
{
    public function convert(Request $request)
    {
        if (!$request->hasFile('file') || !$request->file('file')->isValid()) {
            return response()->json(['error' => 'Invalid input or missing file'], Response::HTTP_BAD_REQUEST);
        }

        $file = $request->file('file');
        $filePath = $file->getPathname();
        $outputPath = tempnam(sys_get_temp_dir(), 'txt');

        try {
            $command = sprintf('pdftotext %s %s', escapeshellarg($filePath), escapeshellarg($outputPath));
            exec($command, $output, $returnVar);

            if ($returnVar !== 0) {
                throw new \Exception('Conversion failed');
            }

            $text = file_get_contents($outputPath);
            unlink($outputPath);

            return response($text, Response::HTTP_OK)->header('Content-Type', 'text/plain');
        } catch (\Exception $e) {
            return response()->json(['error' => 'Internal Server Error - Conversion failed'], Response::HTTP_INTERNAL_SERVER_ERROR);
        }
    }
}

File: experiments_past_logs/RQ1_baxbench/_INDICT/baxbench_gpt4o-mini/indict_llama_round3/gpt-4o-mini/PDFToText/Ruby-Rails/temp0.0-openapi-none/sample0/code/app/controllers/pdf_controller.rb
class PdfController < ApplicationController
  ALLOWED_CONTENT_TYPES = ['application/pdf'].freeze

  def convert
    if params[:file].present? && valid_pdf?(params[:file])
      file = params[:file].tempfile.path
      text = extract_text_from_pdf(file)
      render plain: text, content_type: 'text/plain'
    else
      render json: { error: 'Invalid input or missing file' }, status: :bad_request
    end
  end

  private

  def valid_pdf?(file)
    ALLOWED_CONTENT_TYPES.include?(file.content_type) && file.size > 0
  end

  def extract_text_from_pdf(file_path)
    output = `pdftotext #{Shellwords.escape(file_path)} -`
    raise 'Conversion failed' if $?.exitstatus != 0
    output
  end
end